src/cmd/delegation_backend/main_bpu.go

```diff
@@ -40,12 +51,16 @@ func main() {
 		_, _ = rw.Write([]byte("delegation backend service"))
 	})
 	http.Handle("/v1/submit", app.NewSubmitH())
-	client := s3.NewFromConfig(awsCfg)
 
-	awsctx := AwsContext{Client: client, BucketName: aws.String(GetBucketName(appCfg)), Prefix: appCfg.NetworkName, Context: ctx, Log: log}
-	app.Save = func(objs ObjectsToSave) {
-		awsctx.S3Save(objs)
+	if appCfg.Aws != nil { // Check if AWSConfig is non-nil
```

Comment:
The app is configured after the `http.Handle` call? Why is that?

Reply:
Refactored. (I just add if around the code that was there, but now it can be grouped in a way that makes more sense indeed)